### DISCUSSION TITLE
Question about running tests

### DISCUSSION THREAD

**Discussion by titan73** [The-OpenROAD-Project/OpenROAD#6059] [Q&A]
When running make test and some tests fail I get:
```
Errors while running CTest
Output from these tests are in: /home/costa/OpenROAD/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make: *** [Makefile:71: test] Error 8
```
When should these options "--rerun-failed --output-on-failure" be used? I guess it's for "ctest" such as "ctest --rerun-failed --output-on-failure", correct?

Other questions, is there a way to launch the test for a specific tool such as "ppl" or a specific test?

**Comment by maliberty** [comment]
I usually use 'make test' to run all tests from cmake.  For one dir you can either use the local regression script or use -R with ctest to filter.  I've never seen that particular message.
